### DISCUSSION TITLE
semver.npmjs.com - Please add a note in the caret microdoc/card documenting that caret behavior differs for 0.x versions

### DISCUSSION THREAD

**Discussion by ronjouch** [npm/feedback#137] [Website] [answered]
I was using the excellent [npm semver calculator](https://semver.npmjs.com/) mini-tool to understand why a caret-pinned dependency (axios `^0.19.1`) didn't match the latest release, which bears the same major version (axios `0.21.1`).

It took some googling to figure out that versions 0.something have special handling. As documented in [npm semver docs / caret ranges](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004) (emphasis mine),

> Allows changes that do not modify the left-most ***non-zero*** digit in the `[major, minor, patch]` tuple. In other words, this allows patch and minor updates for versions `1.0.0` and above, patch updates for versions `0.X >=0.1.0`, and _no_ updates for versions `0.0.X`.
> 
> Many authors treat a `0.x` version as if the `x` were the major "breaking-change" indicator.
> 
> Caret ranges are ideal when an author may make breaking changes between `0.2.4` and `0.3.0` releases, which is a common practice. However, it presumes that there will _not_ be breaking changes between `0.2.4` and `0.2.5`. It allows for changes that are presumed to be additive (but non-breaking), according to commonly observed practices.

→ It would be great if the micro-doc at the bottom of the page mentioned this. Here is a suggestion of what it could look like:

![screenshot_npm-semver-3-suggestion_2021-01-05-16:23:01](https://user-images.githubusercontent.com/522085/103703079-eadbb500-4f74-11eb-8aee-dbb75d58968a.png)

Thanks for semver.npmjs.com 🙂

**Comment by MylesBorins** [comment]
hey @ronjouch thanks for the heads up!

I'm digging in with the team right now to figure out where the source of this site lives so we can update it.

**Reply by ljharb** [reply]
https://github.com/npm/semver.npmjs.com/, before it was made private.

**Reply by MylesBorins** [reply]
I've opened a PR internally and am awaiting review.

**Reply by ljharb** [reply]
it'd also be great to make it public again, so the community is able to improve it :-D but that's a different feedback issue, i suppose.

**Reply by MylesBorins** [reply]
I've kicked off a discussion to see why it was made private

**Reply by ronjouch** [reply]
> I've opened a PR internally and am awaiting review.

Hi @MylesBorins, it's been two weeks now 🙂. Gentle ping to ensure this isn't lost internally.

**Reply by MylesBorins** [reply]
Pinged team, thanks for the reminder 👍

**Reply by ronjouch** [reply]
@MylesBorins one more gentle ping, as it's been one month since the last one. I'm removing my calendar event and will stop reminding you about it; if you want to do it, cool, else well I tried.

**Reply by MylesBorins** [reply]
hey @ronjouch still very much on my radar (have a perpetually open tab for both this thread and the open PR I have to update the text. Apologies I haven't been able to get the update landed. Will try and get something together this week.

**Comment by MylesBorins** [comment, marked as answer]
We have deployed a change that fixes this! Thanks for bringing it to our attention

https://semver.npmjs.com/

**Reply by ronjouch** [reply]
Yay! Thaaaaaaaaaaaaaaanks for taking the time to get this done 🙂
